### PR TITLE
Added Display for mode, fixes #13

### LINKED ISSUE #13: Error printed on bad exit status looks ugly
This line prints the error-on-bad-exit-status:

https://github.com/oli-obk/ui_test/blob/944438e6f95c06268d0c628184301af4be2b3dce/src/lib.rs#L333

It uses the debug formatting for `mode`, which can look quite ugly now that this is an enum where some variants have data.

### PR DESCRIPTION
**Pull request by Dajamante**
Fixes #13

### CODE REVIEW

**Comment by oli-obk** [comment]
you also need to run rustfmt

**Comment by Dajamante** [comment]
> 

Sorry, will do tomorrow!

**Comment by oli-obk** [comment]
No rush, I'm just leaving comments as I notice things, address them whenever you were going to work on this anyway. Don't schedule additional time for it.

**Comment by RalfJung** [comment]
Also please edit the PR description to "Fixes #13" so that GitHub will automatically close the issue when the PR lands. :)

File: src/lib.rs [hunk 1]
```diff
@@ -774,3 +774,13 @@ impl Mode {
         }
     }
 }
+
+impl Display for Mode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Mode::Pass => write!(f, "pass"),
+            Mode::Panic => write!(f, "panic"),
+            Mode::Fail{ require_patterns: _ }  => write!(f, "fail ")
```
Comment (RalfJung) [inline src/lib.rs:783]: Why does this have a trailing space, but the other two do not?

Comment (Dajamante) [reply src/lib.rs]: Because I did not run rustfmt. Done now!

**Pull request merged**